tests/phpunit/elementor/modules/atomic-widgets/test-styles-renderer.php

```diff
@@ -0,0 +1,334 @@
+<?php
+
+namespace Elementor\Testing\Modules\AtomicWidgets;
+
+use Spatie\Snapshots\MatchesSnapshots;
+use Elementor\Modules\AtomicWidgets\Styles\Styles_Renderer;
+use ElementorEditorTesting\Elementor_Test_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+    exit; // Exit if accessed directly.
+}
+
+class Test_Styles_Renderer extends Elementor_Test_Base {
+	use MatchesSnapshots;
+
+    public function test_styles_renderer__basic_style() {
+        // Arrange.
+        $styles = [
+            [
+                'id' => 'test-style',
+                'type' => 'class',
+                'variants' => [
+                    [
+                        'props' => [
+                            'color' => 'red',
+                            'fontSize' => '16px',
+                        ],
+                        'meta' => [],
+                    ],
+                ],
+            ],
+        ];
+        $transformers = [];
+        $breakpoints = [];
+
+        $styleRender = new Styles_Renderer( $transformers, $breakpoints);
+
+        // Act.
+        $css = $styleRender->render( $styles );
+
+        // Assert.
+		$this->assertMatchesSnapshot( $css );
+    }
+
+	public function test_styles_renderer__type_invalid_style() {
+		// Arrange.
+		$styles = [
+			[
+				'id' => 'test-style',
+				'type' => 'id', // currently only supports 'class'
+				'variants' => [
+					[
+						'props' => [
+							'color' => 'red',
+							'fontSize' => '16px',
+						],
+						'meta' => [],
+					],
+				],
+			],
+		];
+		$transformers = [];
+		$breakpoints = [];
+
+		$styleRender = new Styles_Renderer( $transformers, $breakpoints);
+
+		// Act.
+		$css = $styleRender->render( $styles );
+
+		// Assert.
+		$this->assertEmpty( $css );
```

Comment:
maybe put 2 styles in the config, and assert that only one of them has been rendered?
(to make sure that we don't break everything when one is invalid)

Reply:
Done